sdk/core/core-rest-pipeline/src/util/tokenCycler.ts

```diff
@@ -204,10 +204,20 @@ export function createTokenCycler(
     // If the tenantId passed in token options is different to the one we have
     // Or if we are in claim challenge and the token was rejected and a new access token need to be issued, we need to
     // refresh the token with the new tenantId or token.
-    const mustRefresh =
-      tenantId !== tokenOptions.tenantId || Boolean(tokenOptions.claims) || cycler.mustRefresh;
+    const hasClaimChallenge = Boolean(tokenOptions.claims);
+    const tenantIdChanged = tenantId !== tokenOptions.tenantId;
```

Comment:
nitpick: can we keep `mustRefresh` defined here so the comment above still applies?